bugbug/models/component.py

```diff
@@ -201,5 +187,152 @@ def is_meaningful(product, component):
             if component in top_components
         }
 
+    def is_meaningful(self, product, component):
+        return product in self.PRODUCTS and component not in ["General", "Untriaged"]
+
+    def get_meaningful_product_components(self, full_comp_tuples, threshold_ratio=100):
+        """ From the given full_comp_tuples iterable of (product, component)
+        tuples, returns the set of tuples which have at least 1% of the most
+        common tuple
+        """
+
+        product_component_counts = Counter(full_comp_tuples).most_common()
+
+        max_count = product_component_counts[0][1]
+        print("Max count", product_component_counts[0])
+        threshold = max_count / threshold_ratio
+        print("Threshold", threshold)
+
+        return set(
+            product_component
+            for product_component, count in product_component_counts
+            if count > threshold
+        )
+
     def get_feature_names(self):
         return self.extraction_pipeline.named_steps["union"].get_feature_names()
+
+    def check(self):
+        success = super().check()
+
+        # Get the number of bugs per full component to fasten up the check
+        bugs_number = get_product_component_count()
+
+        # Check number 1, check that the most meaningful product components
+        # still have at least a bug in this component. If the check is failing
+        # that could mean that:
+        # - A component has been renamed / removed
+        # - A component is not used anymore by developers
+        # - TODO: Complete this list
+
+        for product, component in self.meaningful_product_components:
+            full_comp = f"{product}::{component}"
+
+            if full_comp not in bugs_number.keys():
+                msg = f"Component {component!r} of product {product!r} doesn't exists, failure"
+                print(msg)
```

Comment:
Nit: instead of storing the string in a variable and then printing it, you could print it directly